libcove2/common.py

```diff
@@ -12,32 +12,116 @@
 )
 
 
-def schema_dict_fields_generator(schema_dict):
+def _resolve_ref(value, defs, registry=None):
+    if value["$ref"].startswith("urn:"):
+        subschema = registry.contents(value["$ref"].split("#")[0])
+        if "#/$defs/" in value["$ref"]:
+            name = value["$ref"].split("$defs/")[-1]
+            defs = subschema["$defs"]
+            subschema = defs[name]
+        return subschema
+    elif value["$ref"].startswith("#/$defs/"):
+        name = value["$ref"].split("$defs/")[-1]
+        return defs[name]
+
+
+def _process_items(schema_dict, registry=None, defs=None):
+    items_schema_dicts = []
+    if "oneOf" in schema_dict["items"] and isinstance(
+        schema_dict["items"]["oneOf"], list
+    ):
+        for oneOf in schema_dict["items"]["oneOf"]:
+            items_schema_dicts.append(oneOf)
+    elif "$ref" in schema_dict["items"]:
+        if schema_dict["items"]["$ref"].startswith("urn:"):
+            subschema = registry.contents(schema_dict["items"]["$ref"].split("#")[0])
+            if "#/$defs/" in schema_dict["items"]["$ref"]:
+                name = schema_dict["items"]["$ref"].split("$defs/")[-1]
+                defs = subschema["$defs"]
+                subschema = defs[name]
+            items_schema_dicts.append(subschema)
+        elif schema_dict["items"]["$ref"].startswith("#/$defs/"):
+            name = schema_dict["items"]["$ref"].split("$defs/")[-1]
+            items_schema_dicts.append(defs[name])
+    elif "properties" in schema_dict["items"] and isinstance(
+        schema_dict["items"]["properties"], dict
+    ):
+        items_schema_dicts.append(schema_dict["items"])
+    return items_schema_dicts
+
+
+def schema_dict_fields_generator(schema_dict, registry=None, defs=None):
```

Comment:
Could we have a note in docstring of what type of var registry and defs are expected to be?

Reply:
I'm still slightly confused what exactly registry should be but I'm happy to merge this in and then I'll send you a docs PR and we can fine tune